.github/workflows/haskell.yml

```diff
@@ -62,7 +62,7 @@ jobs:
       id: setup-haskell
       with:
         ghc-version: ${{ matrix.ghc }}
-        cabal-version: 3.4.0.0
+        cabal-version: 3.6.2.0
```

Comment:
Wonder whether the [cabal version in shell.nix](https://github.com/input-output-hk/cardano-ledger/pull/3133/files#diff-c09cc8dc973415b1221308559b96942a1c4b872e690bf360fc22d9b42822d519L65) doesn't need updating as well